panic when creating secret on berglas bucket with wrong permissions

**Steps to reproduce**

1. Create a berglas bucket (my service account doesn't have list/write/read roles to the newly created berglas bucket)
2. Attempt to create a berglas secret using berglas cli
```
➜  project-x git:(master) ✗ berglas create gs://<berglas-bucket>/database-username jonny \
--key projects/<project-id>/locations/global/keyRings/<key-ring>/cryptoKeys/<key-name>
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x152db8b]
goroutine 1 [running]:
github.com/GoogleCloudPlatform/berglas/pkg/berglas.secretFromAttrs(...)
	/private/tmp/berglas-20191013-53234-k2zn8c/berglas-0.2.1/src/github.com/GoogleCloudPlatform/berglas/pkg/berglas/berglas.go:123
github.com/GoogleCloudPlatform/berglas/pkg/berglas.(*Client).encryptAndWrite(0xc0002e2000, 0x17ef480, 0xc0001aa9c0, 0x7ffeefbff944, 0x14, 0x7ffeefbff959, 0x24, 0x7ffeefbff98d, 0x4e, 0xc0001a2610, ...)
	/private/tmp/berglas-20191013-53234-k2zn8c/berglas-0.2.1/src/github.com/GoogleCloudPlatform/berglas/pkg/berglas/writer.go:110 +0x96b
github.com/GoogleCloudPlatform/berglas/pkg/berglas.(*Client).Create(0xc0002e2000, 0x17ef480, 0xc0001aa9c0, 0xc00022bcc8, 0x1, 0xc0002e2000, 0x0)
	/private/tmp/berglas-20191013-53234-k2zn8c/berglas-0.2.1/src/github.com/GoogleCloudPlatform/berglas/pkg/berglas/create.go:75 +0xe7
github.com/GoogleCloudPlatform/berglas/pkg/berglas.Create(0x17ef480, 0xc0001aa9c0, 0xc00022bcc8, 0x8, 0x8, 0x0)
	/private/tmp/berglas-20191013-53234-k2zn8c/berglas-0.2.1/src/github.com/GoogleCloudPlatform/berglas/pkg/berglas/create.go:30 +0xa2
main.createRun(0x1c25fe0, 0xc0001aa980, 0x2, 0x4, 0x0, 0x0)
	/private/tmp/berglas-20191013-53234-k2zn8c/berglas-0.2.1/src/github.com/GoogleCloudPlatform/berglas/main.go:500 +0x285
github.com/spf13/cobra.(*Command).execute(0x1c25fe0, 0xc0001aa940, 0x4, 0x4, 0x1c25fe0, 0xc0001aa940)
	/private/tmp/berglas-20191013-53234-k2zn8c/berglas-0.2.1/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0x1c25360, 0x1c2ef20, 0x1708409, 0x3)
	/private/tmp/berglas-20191013-53234-k2zn8c/berglas-0.2.1/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/private/tmp/berglas-20191013-53234-k2zn8c/berglas-0.2.1/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
	/private/tmp/berglas-20191013-53234-k2zn8c/berglas-0.2.1/src/github.com/GoogleCloudPlatform/berglas/main.go:422 +0xb88
```

**Possible Causes**

I get a panic error most likely because whenever attempting to write to the bucket I don't have necessary permissions. This is most likely due to the fact that there are two error branches that are not being checked 

https://github.com/GoogleCloudPlatform/berglas/blob/master/pkg/berglas/writer.go#L97-L107

```go
		if terr, ok := err.(*googleapi.Error); ok {
			switch terr.Code {
			case http.StatusNotFound:
				return nil, errors.New("bucket does not exist")
			case http.StatusPreconditionFailed:
				if conds.DoesNotExist {
					return nil, errSecretAlreadyExists
				}
				return nil, errSecretModified
                         /** default: // other error codes are not being checked */
			}
		} /** else { // Non-google error is not checked } */
```

--- a/pkg/berglas/writer.go
+++ b/pkg/berglas/writer.go
@@ -103,7 +103,11 @@ func (c *Client) encryptAndWrite(
 					return nil, errSecretAlreadyExists
 				}
 				return nil, errSecretModified
+			default:
+				return nil, errors.Wrap(err, "Writer.Close: googleapi.Error: failed to finalise bucket creation")
 			}
+		} else {
+			return nil, errors.Wrap(err, "Writer.Close: failed to finalise bucket creation")
 		}
 	}
 

